src/components/intel/CampaignSelector/index.js

```diff
@@ -0,0 +1,94 @@
+import { Grid, makeStyles } from "@material-ui/core";
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+import { useCampaignContext } from "components/intel/Home";
+import LabeledSelect from "components/utilities/LabeledSelect";
+
+/* STYLING */
+const useStyles = makeStyles(() => ({
+  select: { minWidth: 200 },
+}));
+
+const CampaignSelector = () => {
+  /* STYLING */
+  const classes = useStyles();
+
+  /* ROUTING */
+  const history = useHistory();
+
+  /* FORM STATE */
+  const [indices, setIndices] = useState({ campaign: -1, world: -1 });
+
+  /* CONTEXT */
+  const { campaigns, activeCampaign, world, setActiveCampaign, setWorld } =
+    useCampaignContext();
+
+  /* FORM STATE */
+  useEffect(() => {
+    setIndices({
+      campaign: campaigns.indexOf(activeCampaign),
+      world: activeCampaign.worlds.indexOf(world),
+    });
+  }, [campaigns, activeCampaign, world]);
+
+  const handleCampaignChange = (event) => {
+    let index = event.target.value;
+    if (index == -1) {
+      return;
+    }
+    history.push(`/campaigns/${campaigns[index].name}`);
+    setActiveCampaign(campaigns[index]);
+  };
+
+  const handleWorldChange = (event) => {
+    let index = event.target.value;
+    if (index == -1) {
+      return;
```

Comment:
could we change this to actually set the active world to -1 and push back to `/campaigns/${activeCampaign.name}`
then add a button that renders when a world is active that calls handle world change with a value of -1?

Reply:
Not sure why on the button. I made it so that if index is -1 then it pushes you back to the campaign page

Reply:
The button so that you can go back to a campaign easy. I find that I end up going back to campaign a lot because I can never remember connections between places.

Reply:
Added